tools/etcd-dump-logs/raw.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"go.uber.org/zap"
+
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/client/pkg/v3/fileutil"
+	"go.etcd.io/etcd/pkg/v3/pbutil"
+	"go.etcd.io/etcd/server/v3/storage/wal"
+	"go.etcd.io/etcd/server/v3/storage/wal/walpb"
+	"go.etcd.io/raft/v3/raftpb"
+)
+
+func readRaw(lg *zap.Logger, fromIndex *uint64, waldir string, out io.Writer) {
+	var walReaders []fileutil.FileReader
+	files, err := ioutil.ReadDir(waldir)
+	if err != nil {
+		lg.Fatal("Failed to read directory.", zap.String("directory", waldir), zap.Error(err))
+	}
+	for _, finfo := range files {
+		if filepath.Ext(finfo.Name()) != ".wal" {
+			lg.Warn("Ignoring not .wal file", zap.String("filename", finfo.Name()))
```

Comment:
```suggestion
			lg.Warn("Ignoring not .wal file", zap.String("filename", finfo.Name()))
			continue
```

Reply:
This comment isn't resolved yet. Did you intentionally include files which do not suffixed with ".wal"?

Reply:
I'm sorry - missed this.   I'm mimicking the logic of the ReadAll that do skips not .wal files.